tests/rptest/services/kerberos.py

```diff
@@ -349,13 +369,61 @@ def metadata(self, principal: str):
         kinit_args = f"-kt {self.keytab_file} -c {client_cache} {principal}"
         kinit_cmd = f"kinit -R {kinit_args} || kinit {kinit_args}"
         sasl_conf = f"-X security.protocol=sasl_plaintext -X sasl.mechanisms=GSSAPI '-Xsasl.kerberos.kinit.cmd={kinit_cmd}' -X sasl.kerberos.service.name=redpanda"
-        res = self.nodes[0].account.ssh_output(
-            cmd=
-            f"KRB5_TRACE=/dev/stderr KRB5CCNAME={client_cache} kcat -L -J -b {self.redpanda.brokers()} {sasl_conf}",
-            allow_fail=False,
-            combine_stderr=False)
-        self.logger.debug(f"Metadata request: {res}")
-        return json.loads(res)
+        try:
+            res = self.nodes[0].account.ssh_output(
+                cmd=
+                f"KRB5_TRACE=/dev/stderr KRB5CCNAME={client_cache} kcat -L -J -b {self.redpanda.brokers()} {sasl_conf}",
+                allow_fail=False,
+                combine_stderr=False)
+            self.logger.debug(f"Metadata request: {res}")
+            return json.loads(res)
+        except RemoteCommandError as err:
+            if b'No Kerberos credentials available' in err.msg:
+                raise AuthenticationError(err.msg) from err
+            raise
+
+    def metadata_java(self, principal: str):
+        self.logger.info("Metadata request (Java)")
+        # TODO: BP - this!
```

Comment:
Oh yeah. I was going to do this.

Reply:
Want me to wait or is going to be addressed in a future PR?